src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -40,6 +42,18 @@ public override async Task<LSP.CompletionItem[]> HandleRequestAsync(LSP.Completi
                 return Array.Empty<LSP.CompletionItem>();
             }
 
+            // C# and VB share the same LSP language server, and thus share the same default trigger characters.
+            // The '{' character, while a trigger character in VB, is not a trigger character in C#, so we must
+            // handle it in a special case.
+            if (request.Context.TriggerCharacter == "{")
```

Comment:
I know I suggested this initially, but I wonder if instead we could look at the trigger characters per language.  Here we import them and ignore language differences - http://sourceroslyn.io/#Microsoft.CodeAnalysis.LanguageServer.Protocol/Handler/Initialize/InitializeHandler.cs,33 - but we could separate them out instead.  THen in here we can check that the character is applicable for the document language.